sdk/core/Azure.Core/src/Shared/DiagnosticScope.cs

```diff
@@ -115,6 +89,11 @@ public void AddLink(string traceparent, string? tracestate, IDictionary<string,
         public void Start()
         {
             Activity? started = _activityAdapter?.Start();
+
+            if (started != null)
+            {
+                this.IsEnabled = IsEnabled && started.GetIsAllDataRequestedFlag();
```

Comment:
Should we just set _currentActivity to null if IsAllDataRequested is false? 
/cc @lmolkova

Reply:
We don't lose the activities due to sampling when we create it using activity source, we just get activities with `IsAllDataRequested` flag off. Hence, I did the same thing here. Otherwise, I am fine with setting activities as null (it would make this struct `readonly `again)

Reply:
But IsEnabled doesn't do anything - it isn't used anywhere.

Reply:
Yes we do! https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/MessagingClientDiagnostics.cs#L173